tests/linearizability/model.go

```diff
@@ -209,5 +231,51 @@ func stepDelete(state EtcdState, request EtcdRequest, response EtcdResponse) (bo
 
 	state.Value = ""
 	state.LastRevision = response.Revision
+	state.FailedWrite = nil
+	return true, state
+}
+
+func stepTxn(state EtcdState, request EtcdRequest, response EtcdResponse) (bool, EtcdState) {
+	if response.Err != nil {
+		state.FailedWrite = &request
+		return true, state
+	}
+	// revision should never decrease
+	if response.Revision < state.LastRevision {
+		return false, state
+	}
+	valueMatch := state.Value == request.TxnExpectData
+
+	if state.FailedWrite == nil {
+		// transaction should succeed if value matches
+		if response.TxnSucceeded != valueMatch {
+			return false, state
+		}
```

Comment:
This isn't correct.  Note that in the `(*readWriteSingleKey) Run`, the `Read` and `Write` operation are not an atomic operation.  Pls see example below, obviously `valueMatch` will be `false`, but it's expected behavior. 

1. client 1 read:  `request.TxnExpectData = readResonse`
2. client 2 write (PUT):  The value changed.
3. client 1 write (TXN): The `txn.If `will be evaluated to `false`